src/components/OverviewMap.js

```diff
@@ -67,74 +67,20 @@ class OverviewMap extends React.Component {
-      const {geojson, id} = hearing;
+      const {geojson} = hearing;
 
       if (geojson) {
-        switch (geojson.type) {
-          case "Polygon": {
-            // XXX: This only supports the _first_ ring of coordinates in a Polygon
-            const latLngs = geojson.coordinates[0].map(([lng, lat]) => new LatLng(lat, lng));
-            contents.push(
-              <Polygon
-                key={Math.random()}
-                positions={latLngs}
-              >
-                {this.getPopupContent(hearing, geojson)}
-              </Polygon>);
-          }
-            break;
-          case "Point": {
-            const latLngs = new LatLng(geojson.coordinates[1], geojson.coordinates[0]);
-            contents.push(
-              <Marker
-                position={latLngs}
-                key={Math.random()}
-                icon={new Leaflet.Icon({
-                  iconUrl: leafletMarkerIconUrl,
-                  shadowUrl: leafletMarkerShadowUrl,
-                  iconRetinaUrl: leafletMarkerRetinaIconUrl,
-                  iconSize: [25, 41],
-                  iconAnchor: [13, 41]
-                })}
-                {...this.getAdditionalParams(hearing)}
-              >{this.getPopupContent(hearing, geojson)}
-              </Marker>
-            );
-          }
-            break;
-          case "LineString": {
-            const latLngs = geojson.coordinates.map(([lng, lat]) => new LatLng(lat, lng));
-            contents.push(
-              <Polyline key={Math.random()} positions={latLngs}>{this.getPopupContent(hearing, geojson)}</Polyline>
-            );
-          }
-            break;
-          // eslint-disable-next-line no-lone-blocks
-          case "FeatureCollection": {
-            // if mapElementLimit is true & more than 0, display up to that amount of elements
-            if (mapElementLimit && mapElementLimit > 0) {
-              geojson.features.slice(0, mapElementLimit).forEach((feature) => {
-                contents.push(this.getMapElement(feature, hearing));
-              });
-            } else {
-              // if mapElementLimit is false -> display all elements found in FeatureCollection
-              geojson.features.forEach((feature) => {
-                contents.push(this.getMapElement(feature, hearing));
-              });
-            }
-          }
-            break;
-          default:
-          // TODO: Implement support for other geometries too (markers, square, circle)
-            contents.push(
-              <GeoJSON data={geojson} key={JSON.stringify(geojson)}>{this.getPopupContent(hearing, geojson)}</GeoJSON>
-            );
+        const mapElement = this.getMapElement(geojson, hearing);
+        if (Array.isArray(mapElement) && mapElement.length > 0) {
+          mapElement.forEach(mapEl => {
+            contents.push(mapEl);
+          });
+        } else if (mapElement && !Array.isArray(mapElement)) {
+          contents.push(mapElement);
```

Comment:
this section could be simplified further, as the else if tests seem redundant. 
![image](https://user-images.githubusercontent.com/50320115/225013570-ffcc9446-fba2-40d6-a671-d5da13486085.png)

Reply:
I felt like adding empty arrays into the contents array would be sort of bad practice even if it currently does work. This way the contents array always only consists of objects that we want to show on the map and we're not reliant on the map library to handle the exceptions.